Makefile

```diff
@@ -1,149 +1,152 @@
+
+
+define bundle-ref
+	$(shell flatpak build-bundle $(1) "$(OUTDIR)/$(subst /,-,$(2)).flatpak" $(2))
 endef
 
-all: sdk icontheme
+define clean-ref
+	$(shell rm -f "$(OUTDIR)/$(subst /,-,$(1)).flatpak")
+endef
 
-sdk: ${REPO} $(SDK_DEPS)
-	$(call build-manifest,com.endlessm.apps.Sdk.json)
+define sign-ref
+	$(shell ostree gpg-sign --repo=$(1) $(2) $(GPG_KEY_ID))
+endef
 
-icontheme: ${REPO} $(ICONTHEME_DEPS)
-	$(call build-manifest,org.freedesktop.Platform.Icontheme.EndlessOS.json)
 
-${REPO}:
-	ostree init --mode=archive-z2 --repo=${REPO}
+all: export
 
-com.endlessm.apps.Sdk.json: com.endlessm.apps.Sdk.json.tmpl generate-manifest.py Makefile
-	@echo "  GEN   $@"; \
-	./generate-manifest.py \
-		--arch=$(ARCH) \
-		--sdk-branch=$(SDK_BRANCH) \
-		--base-runtime-version=$(GNOME_RUNTIME_VERSION) \
-		$< > $@
+check: ;
+.PHONY: check
 
-add-repo:
-	flatpak remote-add --user --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+clean:
+	if [ -d "$(OUTDIR)" ]; then rmdir $(OUTDIR); fi
+	if [ -d "$(CACHEDIR)" ]; then rmdir $(CACHEDIR); fi
+	if [ -d "$(REPO)" ]; then rm -r $(REPO); fi
+.PHONY: clean
 
-install-dependencies: add-repo
-	for dep in $(FDO_DEPS) $(GNOME_DEPS); do \
-		flatpak install -y --user flathub $$dep ; \
-		flatpak update -y --user $$dep ; \
-	done
-	flatpak list --user --runtime --show-details
+export: ;
+.PHONY: export
 
-install-locale-dependencies: add-repo
-	for dep in $(LOCALE_DEPS); do \
-		flatpak uninstall -y --user $$dep ; \
-		flatpak install -y --user flathub $$dep ; \
-	done
-	flatpak list --user --runtime --all --show-details
+sign: export
+.PHONY: sign
 
-clean-dependencies: add-repo
-	flatpak uninstall -y --user $(FDO_DEPS)
-	flatpak uninstall -y --user $(GNOME_DEPS)
+bundle: ;
+.PHONY: bundle
 
-check: com.endlessm.apps.Sdk.json
-	@echo "  CHK   $<"; json-glib-validate com.endlessm.apps.Sdk.json
+fetch-junctions:
+	$(BST) $(BST_ARGS) fetch freedesktop-sdk.bst gnome-sdk.bst
+.PHONY: fetch-junctions
 
-clean:
-	@rm -rf builddir 
-	@rm -f ${SUBST_FILES}
-	@rm -f com.endlessm.apps.Sdk.json
-	@rm -f com.endlessm.apps.Sdk.json.in
-
-maintainer-clean: clean
-	@rm -rf .flatpak-builder
-	@rm -rf ${REPO}
-
-import-artefacts:
-	@if [ -f ${REPO}.tar ]; then \
-	    tar xf ${REPO}.tar && \
-	    ostree fsck --repo=${REPO} ; \
-	    rm -f ${REPO}.tar
-	fi
-
-bundle-artefacts:
-	@tar cf ${REPO}.tar ${REPO}
-
-update-fonts:
-	@for font in $(GOOGLE_FONTS); do \
-		wget https://fonts.google.com/download?family=$$font -O fonts/$$font.zip; \
-	done
-
-.PHONY: sdk icontheme add-repo install-dependencies clean-dependencies maintainer-clean bundle-artefacts
+update-refs:
+	$(BST) $(BST_ARGS) track freedesktop-sdk.bst gnome-sdk.bst
+	$(BST) $(BST_ARGS) track flatpak-runtimes.bst flatpak-platform-extensions.bst --deps=all
+.PHONY: update-refs
+
+
+$(OUTDIR):
+	mkdir -p $(OUTDIR)
+
+$(CACHEDIR):
+	mkdir -p $(CACHEDIR)
+
+$(REPO):
+	$(OSTREE) init --repo=$(REPO) --mode=bare-user-only
```

Comment:
You need this to remain an archive repo or we won't be able to publish from it. I suppose you could precreate the repo in the jenkins script the way we need it there. OSTree just returns immediately if the repo is already initialized.